src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs

```diff
@@ -40,17 +45,12 @@ public static OSPlatform Create(string osPlatform)
 
         public bool Equals(OSPlatform other)
         {
-            return Equals(other._osPlatform);
-        }
-
-        internal bool Equals(string? other)
-        {
-            return string.Equals(_osPlatform, other, StringComparison.Ordinal);
+            return string.Equals(_osPlatform, other._osPlatform, StringComparison.OrdinalIgnoreCase);
         }
 
         public override bool Equals(object? obj)
         {
-            return obj is OSPlatform && Equals((OSPlatform)obj);
+            return obj is OSPlatform osPlatform && Equals(osPlatform);
```

Comment:
😀

Reply:
another idea for an analyzer (if there is no existing compiler error for that already ;) )